Detectors/FIT/FT0/simulation/src/Digits2Raw.cxx

```diff
@@ -0,0 +1,153 @@
+  digTree->SetBranchAddress("FT0Digit", &digArr);
+  Int_t nevD = digTree->GetEntries(); // digits in cont. readout may be grouped as few events per entry
+  std::cout << "Found " << nevD << " events with digits " << std::endl;
+  uint32_t old_orbit = 0;
+  for (Int_t iev = 0; iev < nevD; iev++) {
+    digTree->GetEvent(iev);
+    for (const auto& digit : *digArr) {
+
+      auto mIntRecord = digit.getInteractionRecord();
+      uint32_t current_orbit = mIntRecord.orbit;
+      if (old_orbit != current_orbit) {
+        old_orbit = current_orbit;
+        mNpages = 0;
+      }
+      convertDigits(digit, lut);
+    }
+  }
+}
+
+/*******************************************************************************************************************/
+void Digits2Raw::convertDigits(const o2::ft0::Digit& digit, const o2::ft0::LookUpTable& lut)
+{
+  auto mIntRecord = digit.getInteractionRecord();
+  std::vector<o2::ft0::ChannelData> mTimeAmp = digit.getChDgData();
+  bool is0TVX = digit.getisVrtx();
+  int oldlink = -1;
+  int nchannels = 0;
+  for (auto& d : mTimeAmp) {
+    int nlink = lut.getLink(d.ChId);
+    if (nlink != oldlink) {
+      uint nGBTWords = uint((nchannels + 1) / 2);
+      flushEvent(oldlink, mIntRecord, nGBTWords);
+      oldlink = nlink;
+      nchannels = 0;
+      mNpages++;
+    }
+    mEventData[nchannels].channelID = lut.getMCP(d.ChId);
+    mEventData[nchannels].charge = MV_2_NCHANNELS * d.QTCAmpl;   //7 mV ->16channels
+    mEventData[nchannels].time = CFD_NS_2_NCHANNELS * d.CFDTime; //1000.(ps)/13.2(channel);
+    mEventData[nchannels].is1TimeLostEvent = 0;
+    mEventData[nchannels].is2TimeLostEvent = 0;
+    mEventData[nchannels].isADCinGate = 1;
+    mEventData[nchannels].isAmpHigh = 0;
+    mEventData[nchannels].isDoubleEvent = 0;
+    mEventData[nchannels].isEventInTVDC = is0TVX ? 1 : 0;
+    mEventData[nchannels].isTimeInfoLate = 0;
+    mEventData[nchannels].isTimeInfoLost = 0;
+    int chain = std::rand() % 2;
+    mEventData[nchannels].numberADC = chain ? 1 : 0;
+    //  std::cout << "@@@@ packed GBT " << nlink << " channelID   " << mEventData[nchannels].channelID << " charge " << mEventData[nchannels].charge << " time " << mEventData[nchannels].time << " chain " << mEventData[nchannels].numberADC << std::endl;
+    //  std::cout << "@@@@ digits channelID   " << d.ChId  << " charge " << d.QTCAmpl << " time " << d.CFDTime << std::endl;
+    nchannels++;
+  }
+  if ((nchannels % 2) == 1)
+    mEventData[nchannels] = {};
+}
+
+void Digits2Raw::flushEvent(int link, o2::InteractionRecord const& mIntRecord, uint nGBTWords)
+{
+  // If we are called before the first link, just exit
+  if (link < 0)
+    return;
+  setRDH(link, mIntRecord, nGBTWords);
+  setGBTHeader(link, mIntRecord, nGBTWords);
+  mFileDest.write(reinterpret_cast<char*>(&mRDH), sizeof(mRDH));
+  mFileDest.write(reinterpret_cast<char*>(&mEventHeader), sizeof(mEventHeader));
+  mFileDest.write(reinterpret_cast<char*>(&mEventData), sizeof(mEventData));
```

Comment:
so, here you flushing to the file not ``nGBTWords = (nchannels+1)/2`` 80 bit words but fixed ``sizeof(mEventData[NCHANNELS_FT0] = 208*64`` bits.
I would suggest to do something like that (after writing RDH and your header):
````
uint8_t padding[6] = {0};
if (nchannels&0x1) {
  // fill mEventData[nchannels] with 0s to flag that this is a dummy data
 }
int nchOnPage = 2*nGBTWords;
for (int ich=0;ich<nchOnPage;) { 
  mFileDest.write( (char*)&mEventData[ich++], 5); // write 1st 40 bits
  mFileDest.write( (char*)&mEventData[ich++], 5); // write next 40 bits
  mFileDest.write( padding, 6); // write 0 padding to 16 bytes
 }
}
````

Reply:
FT0 GBT header is 80bit

Reply:
See my comment  [below](https://github.com/AliceO2Group/AliceO2/pull/2339#discussion_r326389757) about the EventHeader

Reply:
Done, please look line 102 RawEventData.h